server/controllers/profile.js

```diff
@@ -170,17 +175,39 @@ exports.savePhoto = asyncHandler(async (req, res, next) => {
     fs.unlinkSync(photo.path);
   });
 
-  const id = mongoose.Types.ObjectId(user._id);
-  const addPics = await Profile.findOneAndUpdate(
-    { userId: id },
-    { $push: { galleryPics: urls } }
-  );
-  if (!addPics) {
+  const profile = await Profile.findOneAndUpdate({userId: user.id}, {profilePic: urls[0]});
+  
+  if (profile) {
+    const oldUrl = profile.profilePic;
+    if (oldUri) {
```

Comment:
```suggestion
    if (oldUrl) {
```

Reply:
@ugur-akin very good catch! The typo threw me off and I was wondering why profilePic validation isn't being performed  😆